octoprint_mrbeam/software_update_information.py

```diff
@@ -1,69 +1,178 @@
+                            "there is a wrong configured config local file - override local file with server file"
+                        )
+                    finally:
+                        # override local file
+                        f.seek(0)
+                        f.write(r.content)
+                        f.truncate()
+
+        except requests.ReadTimeout:
+            _logger.error("timeout while trying to get the update_config file")
+        except IOError as e:
+            _logger.error(
+                "couldn't find/open the file %s %s",
+                join(self._plugin._settings.getBaseFolder("base"), SW_UPDATE_FILE),
+                e,
+            )
+
+        if newfile:
+            _logger.info("new file => set info")
+
+            # inform SoftwareUpdate Pluging about new config
+            sw_update_plugin = self._plugin._plugin_manager.get_plugin_info(
+                "softwareupdate"
+            ).implementation
+            sw_update_plugin._refresh_configured_checks = True
+            sw_update_plugin._version_cache = dict()
+            sw_update_plugin._version_cache_dirty = True
+
+
+def get_update_information(plugin):
+    """
+    Gets called from the octoprint.plugin.softwareupdate.check_config Hook from Octoprint
+    Starts a thread to look online for a new config file
+    sets the config for the Octoprint Softwareupdate Plugin with the data from the config file
+    @param plugin: calling plugin
+    @return: the config for the Octoprint embedded softwareupdate Plugin
+    """
     tier = plugin._settings.get(["dev", "software_tier"])
     beamos_tier, beamos_date = plugin._device_info.get_beamos_version()
-    _logger.info("SoftwareUpdate using tier: %s", tier)
-
-    # The increased number of separate virtualenv for iobeam, netconnectd, ledstrips
-    # will increase the "discovery time" to find those package versions.
-    # "map-reduce" method can decrease lookup time by processing them in parallel
-    return dict(
-        reduce(
-            dict_merge,
-            [
-                _set_info_mrbeam_plugin(plugin, tier, beamos_date),
-                _set_info_mrbeamdoc(plugin, tier),
-                _set_info_netconnectd_plugin(plugin, tier, beamos_date),
-                _set_info_findmymrbeam(plugin, tier),
-                _set_info_mrbeamledstrips(plugin, tier, beamos_date),
-                _set_info_netconnectd_daemon(plugin, tier, beamos_date),
-                _set_info_iobeam(plugin, tier, beamos_date),
-                _set_info_mrb_hw_info(plugin, tier, beamos_date),
-                # _config_octoprint(plugin, tier),
-            ],
-        )
-    )
+    _logger.info("SoftwareUpdate using tier: %s %s", tier, beamos_date)
+    global _softwareupdate_handler
+    if _softwareupdate_handler is None:
+        _softwareupdate_handler = MrBeamSoftwareupdateHandler(plugin, tier, beamos_date)
+        _softwareupdate_handler.look_for_new_config_file()
```

Comment:
If you decide to remove the `MrBeamSoftwareupdateHandler` class, then you can just pass the plugin directly to `look_for_new_config_file(plugin)`.